_posts/2023-09-15-kafka-producer-client-essentials.md

```diff
@@ -0,0 +1,323 @@
+---
+layout: post
+title: "Developing Kafka clients: A simple producer client"
+date: 2023-09-15
+author: paul_mellor
+---
+
+Strimzi simplifies the deployment and management of Kafka clusters in a Kubernetes environment, ensuring a smooth and hassle-free experience.
+But the real fun begins once your cluster is up and running.
+That's when you can start thinking about how to interact with Kafka to stream data.
+In other words, Kafka doing the job it excels at.
+
+In this post, we'll dive into the essentials of developing a Kafka producer client that can send messages to a Strimzi-managed Kafka cluster.
+To illustrate these concepts, we'll walk through a basic example of a self-contained application that generates and produces messages to a specific Kafka topic.
+
+## Getting started with your producer client
+
+The first thing to consider when developing a producer client application is your preferred programming language. 
+
+After you've decided on that, as a bare minimum, your application must be able to connect to the Kafka cluster and use producers to send messages.
+
+Let's break down the essential steps to get started:
+
+1. Start by choosing the Kafka client library that speaks your choice of programming language. We use Java in this post, but you can use Python, .NET, and so on.
```

Comment:
This may be obvious but I wonder if it's worth mentioning the the Java client is the official client that is part of the Apache Kafka project. All other clients are third party and may not have the same features, compatibility guarantees and behavior.